### ISSUE TITLE
[Bug]: 1.18.1 error in console

### ISSUE THREAD

**Issue by thiagogebrimm** [brainsynder-Dev/SimplePets#141]
### Debug Log

https://1drv.ms/u/s!AoQJvjmmU2Srgq0-HUD6G2AdoSX1jA?e=MkbVdi

### What happened?

```
[15:11:38 ERROR]: Could not pass event PostPetHatEvent to SimplePets v5.0-BUILD-116
java.lang.NoSuchMethodError: 'org.bukkit.OfflinePlayer simplepets.brainsynder.api.user.PetUser.getPlayer()'
        at addon.brainsynder.weight.WeightAddon.handlePetWeight(WeightAddon.java:97) ~[?:?]
        at addon.brainsynder.weight.WeightAddon.onHat(WeightAddon.java:154) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor1305.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:75) ~[purpur-api-1.18.1-R0.1-SNAPSHOT.jar:?]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:76) ~[purpur-api-1.18.1-R0.1-SNAPSHOT.jar:git-Purpur-1520]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[purpur-api-1.18.1-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:630) ~[purpur-api-1.18.1-R0.1-SNAPSHOT.jar:?]
        at simplepets.brainsynder.impl.PetOwner.lambda$setPetHat$21(PetOwner.java:520) ~[SimplePets-1.18.1.jar:?]
        at java.util.Optional.ifPresent(Optional.java:178) ~[?:?]
        at simplepets.brainsynder.impl.PetOwner.setPetHat(PetOwner.java:476) ~[SimplePets-1.18.1.jar:?]
        at simplepets.brainsynder.menu.items.list.Hat$1.lambda$run$0(Hat.java:43) ~[SimplePets-1.18.1.jar:?]
        at java.util.Optional.ifPresent(Optional.java:178) ~[?:?]
        at simplepets.brainsynder.menu.items.list.Hat$1.run(Hat.java:42) ~[SimplePets-1.18.1.jar:?]
        at org.bukkit.craftbukkit.v1_18_R1.scheduler.CraftTask.run(CraftTask.java:101) ~[purpur-1.18.1.jar:git-Purpur-1520]
        at org.bukkit.craftbukkit.v1_18_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:483) ~[purpur-1.18.1.jar:git-Purpur-1520]
        at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1574) ~[purpur-1.18.1.jar:git-Purpur-1520]
        at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:519) ~[purpur-1.18.1.jar:git-Purpur-1520]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1497) ~[purpur-1.18.1.jar:git-Purpur-1520]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1285) ~[purpur-1.18.1.jar:git-Purpur-1520]
        at net.minecraft.server.MinecraftServer.lambda$spin$1(MinecraftServer.java:322) ~[purpur-1.18.1.jar:git-Purpur-1520]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

### Up-To-Date

- [X] I am using the latest build

**Comment by brainsynder-Dev** [comment]
By any chance... are you using a custom build of the plugin?
You have debug messages that we do not have in the actual code...

Also... You did not have to post a custom link to the debug file... when you run the `/pet debug` command it gives you a link to send

**Comment by thiagogebrimm** [comment]
I'm using jenkins version.

A oks, is that when using the command the server crashed and it was difficult to find

**Comment by brainsynder-Dev** [comment]
You are 100% sure you downloaded it from our Jenkins? ( https://ci.pluginwiki.us/job/SimplePets_v5/116/ ) 
There are some debug messages we did NOT add in your debug file... For example: `Player PataAssada exists` we did not add that one in...
And that one is spammed in the whole file

**Comment by thiagogebrimm** [comment]
I have it, I downloaded it last night

**Comment by brainsynder-Dev** [comment]
Mind sending the jar file you have of the plugin?

**Comment by thiagogebrimm** [comment]
[SimplePets-1.18.1.jar.zip](https://github.com/brainsynder-Dev/SimplePets/files/7908950/SimplePets-1.18.1.jar.zip)

**Comment by brainsynder-Dev** [comment]
This stumps me... Have you tried to re-download the jar file from the jenkins....
My decompiler keeps throwing errors when I try and de-compile the jar you sent which normally means corrupted jar file

**Comment by thiagogebrimm** [comment]
I am going to try

**Comment by thiagogebrimm** [comment]
Link generator not work 
![image](https://user-images.githubusercontent.com/75040244/151462136-29580764-c2b8-41e0-aa23-679770d7d314.png)

**Comment by thiagogebrimm** [comment]
DEBUG using the new version off jenkins:  https://1drv.ms/u/s!AoQJvjmmU2Srgq1VCWgBSEidUECyYw?e=3UGD3T

**Comment by brainsynder-Dev** [comment]
> Link generator not work ![image](https://user-images.githubusercontent.com/75040244/151462136-29580764-c2b8-41e0-aa23-679770d7d314.png)

Yea i believe its not generating the link because the file is so large due to the debug messages....
Again... there are debug messages we did not add in your file...

Have you tried running a server with Spigot/PaperSpigot possibly something to do with Purpur

**Comment by thiagogebrimm** [comment]
Purpur is more optimized than paper, and there are functions I need in it, I can test it but it wouldn't help me :(

Could you add purpur support?

**Comment by Thatsmusic99** [comment]
Purpur does not provide any significant optimisations over Paper to my knowledge. They used to be since they forked from Tuinity, which recently got merged into Paper itself. I see they use Pufferfish patches too but I am not aware of the patches they provide. (I see disabling timings is an "optimisation" but I really don't see the point of it, coming from someone who works in host support and just asking people to do /timings paste already too much...) - the functionality on the other hand is understandable.

There shouldn't be any need to add support for the server type since the plugin is meant to outright work unless it's torn the server internals beyond recognition. 

That being said as well, I got the plugin working on it. I only found the GUI wasn't working properly; /pet summon worked fine,

```
> version
[11:59:17 INFO]: Current: git-Purpur-1537 (MC: 1.18.1)*
[11:59:17 INFO]: Previous: git-Paper-148 (MC: 1.18.1)
[11:59:17 INFO]: * You are running the latest version
> version SimplePets
[11:59:26 INFO]: SimplePets version 5.0-BUILD-121
[11:59:26 INFO]: Authors: brainsynder and Thatsmusic99
[11:59:26 INFO]: Contributors: traksag
```
